---
title: "Computational Lab Notebook"
author: "Tess"
date: "9/28/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Sept 23, 2020

*Goal:* Plot DBH's from 2016 across all sites in diameters_clean

*Worked* 
ggplot(data = diameters_clean, mapping = aes(x = spp, y = year_2016)) + geom_boxplot() + labs(title = "2016 Diameters by Species", x = "Species" , y = "2016 Diameter (DBH)") + theme_bw() +theme(text = element_text(size = 20)) + theme(axis.text.x = element_text(angle = 90, size= 16))

*Did not work*
ggplot(data = diameters_clean, mapping = aes(x = site, y = year_2016)) + geom_boxplot() + labs(title = "2016 Diameters by Site", x = "Site" , y = "2016 Diameter (DBH)") + theme_bw() +theme(text = element_text(size = 20))
Warning message:
Continuous x aesthetic -- did you forget aes(group=...)? 

also

ggplot(data = diameters_clean, mapping = aes(x = site, y = year_2016, group = spp)) + geom_bar() + labs(title = "2016 Diameters by Site", x = "Site" , y = "2016 Diameter (DBH)") + theme_bw() +theme(text = element_text(size = 20))
Error: stat_count() can only have an x or y aesthetic.
```{r}
ggplot(data = diameters_clean, mapping = aes(x = spp, y = year_2016)) + geom_boxplot() + labs(title = "2016 Diameters by Species", x = "Species" , y = "2016 Diameter (DBH)") + theme_bw() +theme(text = element_text(size = 20)) + theme(axis.text.x = element_text(angle = 90, size= 16))
```

## Sept. 28, 2020
Bergman's rule relates to mammal body sizes and temps - develop a hypothesis about what we would expect to happen to mammalian body sizes if Bergmans rule is true? are there enough data to test this hypothesis in the surveys_complete? what additional data would we like/need?

*Hypothesis* Mammalian size will decrease as climate change continues to warm the earth

*Data wanted/needed* Need climate data, need mammal dimensions


install notes - can be messy and mess things up or install incompletely
install NOAA data, need to go to website to get API key in order to download large amounts of data
paste noaa key into .rprofile found in project file

  print.cache_info httr
 means that code from pkg overrides base r code

lawn_bbox_polygon(c(-114, 32, -115, 31)) %>% view()
code allows you to pull up a chunk of the map by lat and long, use google maps to get lat and long for a place we haven't been - long first then lat

temp_data <- meteo_tidy_ghcnd(stationid = "USW00003145", var = c("tmin", "tmax"))
file min/max dates: 1982-01-01 / 2020-09-30

## September 30, 2020
lm used for linear model in R


## October 5, 2020
-revision management and data manipulation
-signed up for github tesseracrockett (username)
-logged into repository for our class, repository is a place for code of a project, all revisions go here
-commits do = revisions/edits
-fork the code = this makes a copy of the entire repository and saves to personal username, we now have copies of everthing under our usernames on github
-learning to back up changes in R studio to get hub
-move to terminal from console, it's more archaic, allows us to access file system of comp, directly access commands for comp that operating systems typically mask
-use to tell revision management system who we are, in terminal type following to title ourselves for revision and tell it who we are 
rstudio-user@application-2678354-deployment-6956621-jhbrx:/cloud/project$ git config --global user.name "Tess Crockett"
rstudio-user@application-2678354-deployment-6956621-jhbrx:/cloud/project$ git config --global user.name "tessera.crockett@selu.edu"
-adding code to repository 
  workflow = add, commit, push 
    add = tell git what file we've edited/added to
    commit = tell git to save that file, we're committing to the work we've done
    push = put the code in the online repository
-each commit has it's own unique code - on git, anything with green background is new code that has been added, red means it's been deleted
- we have some initial code, we mess with it, then we have code version two, mess with it, code version 3 - this is a bad habit which is dangerous workflow - in git you have a managed workflow, your file name is overwritten and does not change, but you have your initial commit with it's own unique id, and each subsequent commit has it's own as well - you have one file in R, but all previous loads in git 

##projecttwonotes
this works, but dates not in order
```{r}

ggplot(diameters_project_two, mapping = aes(x = date, y= diameter_by_year, group = spp)) +geom_point(size= .2 ) +geom_smooth(method = "lm") + theme_bw() +theme(axis.text.x = element_text(angle = 90, size = 14))
```

this also works
```{r}
ggplot(diameters_project_two, mapping = aes(x = spp, y= diameter_by_year, group = date)) +geom_point(size= .2 ) +geom_smooth(method = "lm", color ="navy", fill = "white") + theme_bw() +theme(axis.text.x = element_text(angle = 90, size = 14))
```

this works best so far
```{r}
ggplot(diameters_project_two, mapping = aes (x=spp, y=diameter_by_year, group = year)) +geom_line() + geom_smooth() + theme_bw() 
```





##October 14, 2020
```{r}
model_fit <-lm(body_depth~color + carapace_length, data = crabs)
summary(model_fit)

```


```{r}
ggplot(crabs, aes(x= carapace_length, y = body_depth, color=color)) + geom_point() + labs(x = "Carapace length (mm)", y= "Body depth (mm)", color = "Crab color") + scale_color_manual(values = c ("blue", "orange")) + geom_smooth(method = "lm") + annotate("text", x = 20, y = 30, label = "R^2 == 0.988", size = 5)
```
check relationship of body depth and carapace length, check with carapace width


```{r cars}
summary(cars)
```
##October 21, 2020
Cran is where you browse packages by topic and provides tools to automatically install all packages and place your own packages for R

Working on tree of life
installed a number of packages using devtools, then use library to load
```{r}
ants <- c("Atta", "ectatomma", "xymmer", "aenictus", "aneuretus")
ants
[1] "Atta"      "ectatomma" "xymmer"    "aenictus"  "aneuretus"
> (resolve_names <- tnrs_match_names(ants))
```
trns = taxonomic name resolution system - assigns unique identifier to each genus, trys to locate a similar taxonimc name, its a matching service - what you think your taxa is called, what tree of life thinks it should be called, according to tree of life
```{r}
tr <- tol_induced_subtree(ott_ids = ott_id(resolved_names) )
```
finds the tree phylogeny that includes all your taxa and locates any ids, prunes out any taxa you didn't ask for

```{r}
plot(tr)
```
shows a subtree for our genera

```{r}
entrez_search(db = "Nucleotide", term ="Ectatomma", api_key = "ee0c603ffa257f4889b45bdc1e25a7eb0408" )
```
pulls all genetic data from genbank(usernmae and password required), but may include others, update
```{r}
ectatomma <- ectaomma <- entrez_search(db = "Nucleotide", term ="Ectatomma AND 2018:2020[PDAT]", retmax=100, api_key = "ee0c603ffa257f4889b45bdc1e25a7eb0408" )
```
narrows by date and limits the max return (only need to do if a lot of data) and assign it to an object
```{r}
ectatomma$count
ectatomma$QueryTranslation
ectatomma$ids
sequences <- entrez_fetch(db = "nuccore", id = ectatomma$ids, rettype = "fasta")

```
load the gene coding, informative, but format not useful, put into a format that is a little more condensed
```{r}
for( seq_start in seq(1,50,100)){
     recs <- entrez_fetch(db="nuccore", id = ectatomma$ids,
                          rettype="fasta", retmax=50, retstart=seq_start)
     cat(recs, file="Ect.seqs.fasta", append=TRUE)
     cat(seq_start+49, "sequences downloaded\r")
 }
```
this allows you to generate an even interval of numbers between the start andstop of a sequence, easier to visualzie, retstart=seq_stop, notes where the download stopped in the case of an internet 

loop over whole vector and see if we can get ids on all the ants. 
```{r}
search_year <-
```
define a function
then input a taxon
what it will do is take each of taxa, search against nucleotide database and see if we can get ids - can't use entrez_search to do this because it doesn't like more than one term, can't put in "ants" - so create our own iterative
```{r}

```

##october 26, 2020
install r global biodiversity information system and leaflet, a mapping system
```{r}
name_suggest("Atta mexicana")
```
gives rgbif key for atta mexicana
```{r}
occurrences <- occ_search(taxonKey = 5035745, limit = 20)
```
gives us data frame of where atta mexicana occurs
```{r}
no_na <-occurrences$data %>% 
+ select(scientificName, decimalLatitude, decimalLongitude) %>% 
+ drop_na()
> no_na
```
add tiles puts political boundaries on the map, addmarkers put basic country and state boundaries, and each marker correspondes to the lat and long found in the data frame for the species 

##November 2, 2020
look at Teague lesson, tidyverse
```{r}
names(cam) <- make.names(names(cam), unique = TRUE)
names(cam)
```
clean up names


```{r}
cam$Species.Name <- str_replace_all(cam$Species.Name, " ", ".")
cam$Common.Name <- str_replace_all(cam$Common.Name, " ", ".")
cam$Subproject <- str_replace_all(cam$Subproject, " ", ".")
```
now pattern matching or regular expressions, look to replace things that match a pattern, remove spaces and change to . do this by column

```{r}
speciesCount <- cam %>% 
+ select(Common.Name, Count, Subproject) %>% 
+ group_by(Common.Name, Subproject) %>% 
+ summarise(totalCount = sum(Count)) %>% 
+ spread(Subproject, totalCount)
```
restructure so species are rows and count is column

```{r}
speciesCount[is.na(speciesCount)] = 0
```
changes NA to 0

now plot
```{r}
library(iNEXT)
> richness <- iNEXT(camDiv, q = 0, datatype = "abundance")
> ggiNEXT(richness) +
+ theme(legend.position = "right") +
+ theme_bw()
```
then use shannon diversity index to build plot using ggiNEXT
```{r}
ggiNEXT(diversity_Shan) +
  theme(legend.position =  "right") +
  theme_bw  
```

now do using simpson diversity index
```{r}
diversity_Simps <- iNEXT(camDiv, q=2, datatype = "abundance")
> ggiNEXT(diversity_Simps) +
+ theme(legend.position = "right") +
+ theme_bw()
```

##November 4, 2020
filter out non animal species count from file on monday
```{r}
junk <-c("Camera.Misfire", "Calibration.Photos", "Camera.Trapper", "Vehicle", "No.Animal", "Human.non-staff")
```
now mass remove the above vector of names
! mean not, != means not equal
```{r}
speciesCount <- speciesCount %>% 
+ filter(!(Common.Name %in% junk))
```
filters out anything in common name that appears in the junk vector
```{r}
speciesCount$Common.Name
```
went from 40 to down to like 34





## Including Plots

You can also embed plots, for example:

```{r pressure, echo=FALSE}
plot(pressure)
```

Note that the `echo = FALSE` parameter was added to the code chunk to prevent printing of the R code that generated the plot.
